pagai/services/pyrog/pyrog.py

```diff
@@ -37,44 +46,67 @@
 """
 
 
-def get_headers():
-    if not PYROG_TOKEN:
-        raise OperationOutcome(
-            "PYROG_TOKEN is missing from environment: cannot get database credentials"
-        )
-    return {"content-type": "application/json", "Authorization": f"Bearer {PYROG_TOKEN}"}
+class PyrogClient:
+    def __init__(self):
+        self.token = self.login()
 
+    def get_headers(self, auth=True):
+        if auth and not self.token:
+            raise OperationOutcome(
+                "PyrogClient is not authenticated (login has probably failed, check your logs)"
+            )
+        headers = {"content-type": "application/json"}
+        if auth:
+            headers["Authorization"] = f"Bearer {self.token}"
 
-def run_graphql_query(graphql_query, variables=None):
-    """
-    This function queries a GraphQL endpoint
-    and returns a json parsed response.
-    """
-    if not PYROG_URL:
-        raise OperationOutcome(
-            "PYROG_URL is missing from environment: cannot get database credentials"
-        )
+        return headers
 
-    try:
-        response = requests.post(
-            PYROG_URL, headers=get_headers(), json={"query": graphql_query, "variables": variables}
-        )
-    except requests.exceptions.ConnectionError:
-        raise OperationOutcome("Could not connect to the Pyrog service")
+    def run_graphql_query(self, graphql_query, variables=None, auth=True):
```

Comment:
Can you add just one small line to explain the auth argument?
Also, I think something like `auth_needed` or `protected` or I don't know what would make it more understandable

Reply:
that's better indeed! done.